lib/parser.js

```diff
@@ -192,52 +238,104 @@ function getValidator(version) {
   return validate;
 }
 
-async function customDocumentOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options) {
+async function customDocumentOperations(
+  parsedJSON,
+  asyncapiYAMLorJSON,
+  initialFormat,
+  options
+) {
   validateServerVariables(parsedJSON, asyncapiYAMLorJSON, initialFormat);
-  validateServerSecurity(parsedJSON, asyncapiYAMLorJSON, initialFormat, SPECIAL_SECURITY_TYPES);
+  validateServerSecurity(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    SPECIAL_SECURITY_TYPES
+  );
 
   if (!parsedJSON.channels) return;
 
   validateTags(parsedJSON, asyncapiYAMLorJSON, initialFormat);
   validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat);
-  validateOperationId(parsedJSON, asyncapiYAMLorJSON, initialFormat, OPERATIONS);
-
-  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
-  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
+  validateOperationId(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    OPERATIONS
+  );
+
+  await customComponentsMsgOperations(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    options
+  );
+  await customChannelsOperations(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    options
+  );
 }
 
-async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload) {
+async function validateAndConvertMessage(
+  msg,
+  originalAsyncAPIDocument,
+  fileFormat,
+  parsedAsyncAPIDocument,
+  pathToPayload
+) {
   //check if the message has been parsed before
-  if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true) return;
-  const defaultSchemaFormat = getDefaultSchemaFormat(parsedAsyncAPIDocument.asyncapi);
+  if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true)
+    return;
+  const defaultSchemaFormat = getDefaultSchemaFormat(
+    parsedAsyncAPIDocument.asyncapi
+  );
   const schemaFormat = msg.schemaFormat || defaultSchemaFormat;
 
-  await PARSERS[String(schemaFormat)]({
+  const parser = PARSERS[String(schemaFormat)];
+
+  if (!parser) {
+    throw new ParserError({
+      type: 'schema-parser-not-registered',
+      title: 'The specification has an unsupported schema format',
+      detail:
+        'You shall not pass, no schema parser available that can parse this schema. The good news is that we have one for Avro, just install @asyncapi/avro-schema-parser and use registerSchemaParser to register it with the current parser',
+      validationErrors: groupValidationErrors(
+        'servers',
+        'server does not have a corresponding schema object',
+        new Map(),
```

Comment:
this is a map of errors, so definitely cannot be empty.

you choose to use `groupValidationErrors`, which is much better from initial implementation. 

Why?

because you will not throw error immediately when it is found. The intention with `groupValidationErrors` is to use it in error when you want to throw it after thing through all messages. 

without such approach, you would get error for one message, then you fix, then error for another message, and fix, and so on, not so nice.

JSDoc say this about the map
```
map of error elements cause the validation error might happen in many places in the document. 
The key should have a path information where the error was found, the value holds information about error element but it is not mandatory
 ```

so the key is the json path pointing to the affected place in the document. Later it ends up in error object under `location.jsonPointer`. You will need to use `tilde` util for sure. 

So from your test example, 

- first you need to make `validateAndConvertMessage` aware of the channel name and operation where validated message is located. 
- Then in this map, the key would be something like `tilde(${channelName})/${operationName}/message/schemaFormat
- and value is, I don't remember but probably broken value 🤷🏼

does it make sense?